src/pages/ReportAvatar.tsx

```diff
@@ -0,0 +1,58 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React from 'react';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import {withOnyx} from 'react-native-onyx';
+import AttachmentModal from '@components/AttachmentModal';
+import Navigation from '@libs/Navigation/Navigation';
+import type {AuthScreensParamList} from '@libs/Navigation/types';
+import * as ReportUtils from '@libs/ReportUtils';
+import * as UserUtils from '@libs/UserUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import ROUTES from '@src/ROUTES';
+import type SCREENS from '@src/SCREENS';
+import type {Policy, Report} from '@src/types/onyx';
+
+type ReportAvatarOnyxProps = {
+    report: OnyxEntry<Report>;
+    isLoadingApp: OnyxEntry<boolean>;
+    policies: OnyxCollection<Policy>;
+};
+
+type ReportAvatarProps = ReportAvatarOnyxProps & StackScreenProps<AuthScreensParamList, typeof SCREENS.REPORT_AVATAR>;
+
+function ReportAvatar({report = {} as Report, policies, isLoadingApp = true}: ReportAvatarProps) {
+    const policy = policies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID ?? '0'}`];
+    const isArchivedRoom = ReportUtils.isArchivedRoom(report);
+    const policyName = isArchivedRoom ? report?.oldPolicyName : policy?.name;
```

Comment:
We should have used `ReportUtils.getPolicyName(report, false, policy)` to get the policy name. This caused https://github.com/Expensify/App/issues/35262